cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -736,8 +942,19 @@ where
     fn generate_target_block(
         &mut self,
         builder: &mut FunctionBuilder,
+        source_block: Block,
     ) -> Result<(Block, Vec<Value>)> {
-        let block_targets = &self.resources.blocks[1..];
+        // We try to mostly generate forward branches to avoid generating an excessive amount of
+        // infinite loops. But they are still important, so give them a 0.1% chance of existing.
+        let generate_backwards = self.u.ratio(1, 1000)?;
+        let (backwards_blocks, forward_blocks) = self.resources.partition_blocks(source_block);
+        let block_targets = if generate_backwards && !backwards_blocks.is_empty() {
```

Comment:
It's a minor thing, but I'd prefer to not consume any fuzz input if there aren't any backwards blocks to jump to:

```suggestion
        let (backwards_blocks, forward_blocks) = self.resources.partition_blocks(source_block);
        let block_targets = if !backwards_blocks.is_empty() && self.u.ratio(1, 1000)? {
```